### PR TITLE
Reader recommendations: hide placeholders if there are no more results available

### LINKED ISSUE #3138: Reader: Recommendations placeholders don't go away
Last night I noticed that when I go to `/recommendations`, and then scroll to the bottom, that the placeholders don't go away.

I initially noticed in the Windows WordPress.com app, but I also noticed the issue in Chrome as well. Console in Chrome doesn't show any errors.

screenshot 12
screenshot 13

cc @blowery

### PR DESCRIPTION
**Pull request by bluefuton**
At the moment, we try and fetch 100 recommendations from the `/read/recommendations/mine` endpoint and stop displaying loading placeholders after 100 have been loaded.

Unfortunately, the endpoint doesn't always seem to have 100 recommendations to hand. In these cases, the placeholders stick around at the bottom of the list.

This PR adds a check to see if the current page of results contains any recommendations. If it doesn't, we declare that it's the last page, which stops placeholders from being loaded.

Before:

screen shot 2016-02-19 at 11 38 34

After: 

screen shot 2016-02-19 at 11 38 41

### To test

Visit http://calypso.localhost:3000/recommendations and scroll to the bottom of the list. Ensure there are no placeholders present.

For those testing within Automattic, you can force the endpoint to return less than 100 results by modifying the `WPCOM_JSON_API_Blogs_You_May_Like_Endpoint` to only return results for the first page:

```
if ( $exclude_blog_ids ) {
	$response['blogs'] = []; // For testing purposes, don't return any results after the first page
}
```

Fixes #3138.

### CODE REVIEW

File: client/reader/recommendations/for-you/index.jsx [hunk 1]
```diff
@@ -132,7 +132,7 @@ const RecommendedForYou = React.createClass( {
 				<InfiniteList
 					items={ this.state.recommendations }
 					fetchingNextPage={ this.state.fetching }
-					lastPage={ this.state.recommendations && this.state.recommendations.length >= 100 }
+					lastPage={ RecommendedSites.isLastPage() || this.state.recommendations.length >= 100 }
```
Comment (blowery) [inline client/reader/recommendations/for-you/index.jsx]: can we make passing 100 recs set last page in the store?

**Comment by blowery** [comment]
If we consolidate the lastPage info into the store, ship it

**Comment by bluefuton** [comment]
> If we consolidate the lastPage info into the store, ship it

:+1: Moved into store in c495f16.

**Comment by blowery** [comment]
:ship:

**Pull request merged**